src/com/lilithsthrone/game/combat/Attack.java

```diff
@@ -495,20 +493,57 @@ public static float getModifiedDamage(GameCharacter attacker, GameCharacter defe
 			}
 
 		}
+		return damage;
+	}
 
+	public static float getModifiedDamageRacial(GameCharacter attacker, GameCharacter defender, Attack attackType, DamageType damageType, float damage) {
 		if (attacker != null && defender!=null) {
 			// Modifiers based on race resistance:
 			if(!defender.hasStatusEffect(StatusEffect.DESPERATE_FOR_SEX)) {
 				damage *= 1 - Util.getModifiedDropoffValue(defender.getAttributeValue(attacker.getRace().getResistanceMultiplier()), 100)/100f;
 			}
 			// Modifiers based on race damage:
 			damage *= 1 + Util.getModifiedDropoffValue(attacker.getAttributeValue(defender.getRace().getDamageMultiplier()), 100)/100f;
-			
+		}
+
+		return damage;
+	}
+
+	public static float getModifiedDamageLevel(GameCharacter attacker, GameCharacter defender, Attack attackType, DamageType damageType, float damage) {
+		if (attacker != null && defender!=null) {
 			// Modifiers based on level:
 			float levelBoost = (attacker.getLevel() - defender.getLevel())*2;
 			levelBoost = Util.getModifiedDropoffValue(levelBoost, 100)/100f;
 			damage = damage * (1 + (levelBoost/100));
 		}
+
+		return damage;
+	}
+	
+	/**
+	 * Applies attacker and defender resistances and bonuses to the supplied attack value, then returns the result.
+	 * 
+	 * 
+	 * @param attacker
+	 *            The attacking character.
+	 * @param defender
+	 *            The defending character.
+	 * @param attackType
+	 *            Type of this attack.
+	 * @return Modified damage value.
+	 */
+	public static float getModifiedDamage(GameCharacter attacker, GameCharacter defender, Attack attackType, DamageType damageType, float attackersDamage) {
+		float damage = attackersDamage;
+		
+		if( 0 == Math.signum( getModifiedDamageImmunity(attacker, defender, attackType, damageType, damage) ) )
```

Comment:
this is still beyond silly, it makes me mentally trip over myself every time I read it.